### DISCUSSION TITLE
PixiJS assets syntax replacing deprecated loader

### DISCUSSION THREAD

**Discussion by ripmurdock** [pixijs/pixijs#9472] [Q&A]
Hi,

I'm attempting to update the PixiJS used in this codepen example to PixiJS 6 or 7.
https://codepen.io/ripmurdock/pen/NWEGXvy?editors=0010

Is there a simple syntax change that can make the deprecated loader lines compatible?

It's referenced here and passes 2 parameters to the init function:

var loader = new PIXI.loaders.Loader(baseUrl)
  .add("face", "awesome250.png")  
  .load(init);

function init(loader, assets) {
  texture = assets.face.texture;
}

**Comment by jasonsturges** [comment]
### Texture

If all you need is a texture, I'd imagine the most simplistic form:

```js
const t = Texture.from("https://s3-us-west-2.amazonaws.com/s.cdpn.io/106114/awesome250.png");
const s = Sprite.from(t);
viewport.addChild(s)
```

<img width="931" alt="image" src="https://github.com/pixijs/pixijs/assets/1213591/e0ebfc52-14f1-4fc1-9842-045a0670885b">


---

### Async / await

Or with Assets, preload with async / await:

```js
async function init()
{
    const texture = await Assets.load("https://s3-us-west-2.amazonaws.com/s.cdpn.io/106114/awesome250.png");
    const face = Sprite.from(texture);
    viewport.addChild(face);
}

init();
```

<img width="949" alt="image" src="https://github.com/pixijs/pixijs/assets/1213591/efc0a804-cc82-42ef-a00d-fe4588e60163">


---

### Promise with Alias

Preload with a promise, and alias the asset as "face" for referencing.

```js
Assets.add(
  "face",
  "https://s3-us-west-2.amazonaws.com/s.cdpn.io/106114/awesome250.png"
);

const promises = Assets.load(["face"]);

promises.then((textures) => {
  const face = Sprite.from(textures.face);
  viewport.addChild(face);
});
```

<img width="934" alt="image" src="https://github.com/pixijs/pixijs/assets/1213591/4711d9c5-5d5b-448f-a085-7a63b92273ac">


---

### Other references

At one time I'd seen an alternate syntax for that, which doesn't seem to work:

```js
Assets.add({
  alias: "face",
  src: "https://s3-us-west-2.amazonaws.com/s.cdpn.io/106114/awesome250.png"
});

const face = await Assets.load('face');
```
